xchange-ascendex/src/main/java/org/knowm/xchange/ascendex/AscendexAdapters.java

```diff
@@ -0,0 +1,255 @@
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.marketdata.Trade;
+import org.knowm.xchange.dto.marketdata.Trades;
+import org.knowm.xchange.dto.meta.CurrencyMetaData;
+import org.knowm.xchange.dto.meta.CurrencyPairMetaData;
+import org.knowm.xchange.dto.meta.ExchangeMetaData;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.dto.trade.OpenOrders;
+import org.knowm.xchange.dto.trade.UserTrade;
+import org.knowm.xchange.dto.trade.UserTrades;
+import org.knowm.xchange.utils.jackson.CurrencyPairDeserializer;
+
+import java.time.Instant;
+import java.util.*;
+
+public class AscendexAdapters {
+
+  public static OrderBook adaptOrderBook(AscendexOrderbookDto ascendexOrderbookDto) {
+    List<LimitOrder> asks = new ArrayList<>();
+    List<LimitOrder> bids = new ArrayList<>();
+
+    CurrencyPair currencyPair =
+        CurrencyPairDeserializer.getCurrencyPairFromString(ascendexOrderbookDto.getSymbol());
+
+    ascendexOrderbookDto
+        .getData()
+        .getAsks()
+        .forEach(
+            askOrder ->
+                asks.add(
+                    new LimitOrder.Builder(Order.OrderType.ASK, currencyPair)
+                        .limitPrice(askOrder.getPrice())
+                        .originalAmount(askOrder.getVolume())
+                        .build()));
+
+    ascendexOrderbookDto
+        .getData()
+        .getBids()
+        .forEach(
+            bidOrder ->
+                bids.add(
+                    new LimitOrder.Builder(Order.OrderType.BID, currencyPair)
+                        .limitPrice(bidOrder.getPrice())
+                        .originalAmount(bidOrder.getVolume())
+                        .build()));
+
+    return new OrderBook(ascendexOrderbookDto.getData().getTimestamp(), asks, bids);
+  }
+
+  public static AccountInfo adaptAccountInfo(
+      List<AscendexCashAccountBalanceDto> ascendexCashAccountBalanceDtoList) {
+    List<Balance> balances = new ArrayList<>();
+
+    ascendexCashAccountBalanceDtoList.forEach(
+            ascendexCashAccountBalanceDto ->
+            balances.add(
+                new Balance.Builder()
+                    .currency(new Currency(ascendexCashAccountBalanceDto.getAsset()))
+                    .available(ascendexCashAccountBalanceDto.getAvailableBalance())
+                    .total(ascendexCashAccountBalanceDto.getTotalBalance())
+                    .frozen(
+                        ascendexCashAccountBalanceDto
+                            .getTotalBalance()
+                            .subtract(ascendexCashAccountBalanceDto.getAvailableBalance()))
+                    .build()));
+
+    return new AccountInfo(
+        Wallet.Builder.from(balances)
+            .id("spot")
+            .features(new HashSet<>(Collections.singletonList(Wallet.WalletFeature.TRADING)))
+            .build());
+  }
+
+  public static AscendexPlaceOrderRequestPayload adaptLimitOrderToBitmaxPlaceOrderRequestPayload(
```

Comment:
bitmax still here, theres 136 references of bitmax throughout the code

Reply:
Fixed